strawberry_django/relay/list_connection.py

```diff
@@ -69,6 +113,58 @@ def resolve_connection(
                 conn.nodes = nodes
                 return conn
 
+        if queryset_config and queryset_config.optimized:
+            assert isinstance(nodes, models.QuerySet)
```

Comment:
Maybe move this to a separate `resolve_optimized_connection`, like the `if` above?

And then rename the existing `resolve_connection_from_cache` to `resolve_optimized_connection_by_prefetch`